low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -410,77 +436,92 @@ def _generate_host_update(self):
                 self._host_memory_usage_threshold = float(self._host_memory_usage_threshold)
         except ValueError:
             logger.warning("Host Memory Alert, Invalid host_memory_usage_threshold value are entered in config.")
-            # Assigning default value to _disk_usage_threshold
+            # Assigning default value to _memory_usage_threshold
             self._host_memory_usage_threshold = self.DEFAULT_HOST_MEMORY_USAGE_THRESHOLD
 
+        memory_persistent_data = self.read_persistent_data('MEMORY_USAGE_DATA')
+        if memory_persistent_data.get('memory_usage_time_map') is not None:

```

Comment:
what happens if `memory_persistent_dat`a is None? `memory_persistent_data.get('memory_usage_time_map')` will throw exception.

Reply:
I was using `memory_persistent_data['memory_usage_time_map']` which was throwing keyError when it was None. 
`memory_persistent_data['memory_usage_time_map']` will return None if key is not present and I am checking not None case so it wont throw an exception. 
Checked similar issue after @mariyappanp comment on this PR https://github.com/Seagate/cortx-monitor/pull/381